### PR TITLE
Add integration tests for Organizations Meta Attributes GET API

### PR DESCRIPTION
**Pull request by BimsaraBodaragama**
### Description

> This PR introduces integration tests for the Organizations Meta Attributes pagination functionality. It includes tests for validating the correct behavior of pagination, handling edge cases, and proper error handling when invalid inputs are provided.

### Purpose

> The purpose of this PR is to enhance test coverage for the meta attributes pagination feature in the Organizations API by implementing comprehensive tests that verify correct pagination behavior and error responses for edge cases.

### Goals

> - Validate the correct functionality of pagination for meta attributes, including forward and backward navigation.
> - Ensure appropriate error responses for invalid inputs, such as invalid pagination limits and cursor values.
> - Improve the overall stability and reliability of the API by covering both standard and edge cases related to meta attributes.

### Approach

> 1. Success Tests:
> - Added tests for creating meta attributes and validating pagination with different limits (1, 2, 3, 5, 10, 13) and edge case limits.
> - Implemented forward and backward pagination tests for meta attributes, ensuring proper handling of next and previous links.
> - Included validation of attributes' order across pages, based on various limit values, ensuring consistency with the expected order (Unicode Character order).
>
> 2. Failure Tests:
> - Tested invalid pagination limits (e.g., -1) and verified that the API returns a 400 Bad Request response.
> - Added tests for invalid after and before cursor values to ensure proper error handling with 400 responses.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 1]
```diff
@@ -1482,6 +1709,19 @@ private void addEmailDomainsToOrganization(String organizationId, String... doma
         validateHttpStatusCode(response, HttpStatus.SC_CREATED);
     }
 
+    private String getMetaAttributesEndpoint(boolean isForward, int limit, String after, String before) {
+
+        if (isForward) {
+            return ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_META_ATTRIBUTES_API_PATH + QUESTION_MARK +
+                    LIMIT_QUERY_PARAM + EQUAL + limit + AMPERSAND + RECURSIVE_QUERY_PARAM + EQUAL + false +
+                    (after != null ? AMPERSAND + AFTER_QUERY_PARAM + EQUAL + after : "");
```
Comment (HasiniSama) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1717]: Can we add the base endpoint into a common variable, instead of having it repeated? 

```
String baseEndpoint = ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_META_ATTRIBUTES_API_PATH + 
                          QUESTION_MARK + LIMIT_QUERY_PARAM + EQUAL + limit + 
                          AMPERSAND + RECURSIVE_QUERY_PARAM + EQUAL + false;
if (isForward) {
    return baseEndpoint + (after != null ? AMPERSAND + AFTER_QUERY_PARAM + EQUAL + after : "");
} else {
     return baseEndpoint + (before != null ? AMPERSAND + BEFORE_QUERY_PARAM + EQUAL + before : "");
}
```

Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with https://github.com/wso2/product-is/pull/21054/commits/d1921f987aae0800b4ba25ddc21c7274bca95774

**Comment by BimsaraBodaragama** [comment]
All the tests passed locally:
image
_Source: target/surefire-reports/emailable-report.html_

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 2]
```diff
@@ -1135,6 +1139,208 @@ public void testDeleteOrganizationsForPagination() {
...
 
+    @Test(groups = "organizationMetaAttributesPaginationTests", dependsOnMethods =
+            "testDeleteOrganizationsForPagination")
+    public void createOrganizationsForMetaAttributesPaginationTests() throws JSONException {
+
+        organizations = createOrganizations(1);
+        assertEquals(organizations.size(), 1);
+
+        organizationForMetaAttributes = organizations.get(0);
+        assertNotNull(organizationForMetaAttributes);
```
Comment (HasiniSama) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1150]: Do we need to create a separate `organizationForMetaAttributes` Map here? Can't we directly use `organizations.get(0)` since we use `organizationForMetaAttributes` in only a few places?

For example:

`String organizationId = organizations.get(0).get("ORGANIZATION_ID");
`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 3]
```diff
@@ -1135,6 +1139,208 @@ public void testDeleteOrganizationsForPagination() {
...
+            String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + "/" + organizationId;
+            Response response = getResponseOfPatch(endpointURL, requestBody);
+
+            validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+            // Save the attributes to the instance variable.
+            Map<String, String> attrMap = new HashMap<>();
+            attrMap.put(ORGANIZATION_ID_ATTRIBUTE, organizationId);
+            attrMap.put(ORGANIZATION_META_ATTRIBUTE_ATTRIBUTE, attribute);
+            metaAttributes.add(attrMap);
```
Comment (HasiniSama) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1182]: Do we need a map to store the meta attributes, since all of them are under one organization ID? We could simply store them in a string array.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 4]
```diff
@@ -1135,6 +1139,208 @@ public void testDeleteOrganizationsForPagination() {
...
+                        LIMIT_QUERY_PARAM + EQUAL + limit + AMPERSAND + RECURSIVE_QUERY_PARAM + EQUAL + false;
+        Response response = getResponseOfGetWithOAuth2(endpointURL, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+        List<String> attributes = response.jsonPath().getList(ORGANIZATION_MULTIPLE_META_ATTRIBUTE_ATTRIBUTES);
+        Assert.assertEquals(attributes.size(), limit);
+
+        // Validate the order of the returned attributes.
+        for (int i = 0; i < limit; i++) {
+            Assert.assertEquals(attributes.get(i), metaAttributes.get(i).get(ORGANIZATION_META_ATTRIBUTE_ATTRIBUTE));
```
Comment (HasiniSama) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1215]: Here we can directly use `metaAttributes.get(i)` if a string array list is used.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 2 above)
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with https://github.com/wso2/product-is/pull/21054/commits/7f95c8efe6686a47acfec90aa0973af6e6e3bcc7

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 3 above)
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with https://github.com/wso2/product-is/pull/21054/commits/7f95c8efe6686a47acfec90aa0973af6e6e3bcc7

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 4 above)
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with https://github.com/wso2/product-is/pull/21054/commits/7f95c8efe6686a47acfec90aa0973af6e6e3bcc7

**Comment by BimsaraBodaragama** [comment]
All the tests passed locally:
image
_Source: target/surefire-reports/emailable-report.html_

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 5]
```diff
@@ -773,8 +781,9 @@ public Object[][] organizationPaginationNonNumericEdgeCasesOfLimitProvider() {
             dataProvider = "organizationPaginationNonNumericEdgeCasesOfLimitDataProvider")
     public void testGetPaginatedOrganizationsForNonNumericEdgeCasesOfLimit(String limitQueryParam) {
 
-        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + RECURSIVE_QUERY_PARAM + EQUAL + FALSE +
-                limitQueryParam;
+        String endpointURL =
```
Comment (SujanSanjula96) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:784]: fix formatting

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 6]
```diff
@@ -1125,7 +1135,182 @@ public void testDisableEmailDomainDiscovery() {
         Assert.assertFalse(isEnabled, "Email domain discovery was not successfully disabled.");
     }
 
-    @Test(dependsOnGroups = "organizationDiscoveryPaginationTests")
+    @Test(groups = "organizationMetaAttributesPaginationTests",
+            dependsOnGroups = "organizationDiscoveryPaginationTests")
+    public void testAddMetaAttributesToOrganizations() {
```
Comment (SujanSanjula96) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1140]: Lets add a new line after method signature

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 7]
```diff
@@ -265,7 +272,7 @@ public void testGetOrganization() {
     @DataProvider(name = "dataProviderForFilterOrganizations")
     public Object[][] dataProviderForFilterOrganizations() {
 
-        return new Object[][] {
+        return new Object[][]{
```
[the remark this answers is not in the document]
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with [2122b95](https://github.com/wso2/product-is/pull/21054/commits/2122b959d805379690765472e0f1c6ae56087016)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 5 above)
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: Addressed with [2122b95](https://github.com/wso2/product-is/pull/21054/commits/2122b959d805379690765472e0f1c6ae56087016)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 6 above)
Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:1135]: Addressed with [2122b95](https://github.com/wso2/product-is/pull/21054/commits/2122b959d805379690765472e0f1c6ae56087016)

**Review by SujanSanjula96 — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java [hunk 8]
```diff
@@ -454,4 +454,33 @@ public void testGetPaginatedOrganizationsDiscoveryWithInvalidOffsetAndLimitZero(
         validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, ERROR_CODE_SERVER_ERROR);
     }
 
+    @Test(dependsOnMethods = "testGetPaginatedOrganizationsDiscoveryWithInvalidLimitAndOffset")
+    public void testGetPaginatedMetaAttributesWithInvalidLimit() {
+
+        String invalidLimitUrl =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_META_ATTRIBUTES_API_PATH + QUESTION_MARK +
+                        LIMIT_QUERY_PARAM + EQUAL + "-1";
```
Comment (ShanChathusanda93) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java:462]:
```
                ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_META_ATTRIBUTES_API_PATH + QUESTION_MARK +
                        LIMIT_QUERY_PARAM + EQUAL;
```
There are multiple places where we use this combination. Shall we create a single constant for this?

Comment (BimsaraBodaragama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java:462]: Addressed in PR:
  - https://github.com/wso2/product-is/pull/26155
